### DISCUSSION TITLE
Unable to compile whole project & hence deploy

### DISCUSSION THREAD

**Discussion by truptisatsangi** [zkSync-Community-Hub/zksync-developers#631] [Dev Tooling] [answered]
While using zksync plugin in remix,  I am unable to compile whole project as this error is showing 
"**Contracts folder not shown in workspace**".

 Although I don't think this should be the case as there is contracts folder in the project. I am able to compile single contract file but on deployment I am getting this thing 
 "**No contracts ready for deployment yet, compile a solidity contract**"

**Comment by alhenaai** [comment, marked as answer]
It seems like you are encountering issues with the Remix IDE while trying to compile and deploy contracts using the ZKsync plugin. Here are a few steps you can follow to troubleshoot and resolve the issue:

1. **Ensure the Contracts Folder is Visible in Remix:**
   - Make sure that the `contracts` folder is indeed part of the workspace in Remix. Sometimes, folders might not be properly loaded or visible due to browser cache issues. Try refreshing the page or restarting the browser.

2. **Check the Workspace Settings:**
   - Verify that the workspace settings in Remix are correctly set to include the `contracts` folder. You might need to re-add the folder to the workspace if it's not being recognized.

3. **Compile the Project:**
   - Before deploying, ensure that all contracts within the `contracts` folder are compiled. You can do this by selecting each contract individually in Remix and compiling them.

4. **Deployment Readiness:**
   - After successful compilation, make sure that the compiled contracts are ready for deployment. This typically means ensuring that the contract artifacts (like ABI and bytecode) are generated and visible in the Remix environment.

If you continue to face issues, it might be helpful to clear your browser's cache or try accessing Remix in a different browser to see if the issue persists. Additionally, double-check that the structure of your project in the file explorer of Remix matches what you expect, with all contract files properly placed within the `contracts` folder.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
@truptisatsangi there was an issue with contracts not being deployed if they weren't in the `contracts` folder and it's covered here in this video of how to resolve it if the bot response above didn't help https://youtu.be/WIsXSwo4iDc?si=p-SS7LoqjVkaiAdw&t=16142
